modules/Hoot/ui/modals/importDataset.js

```diff
@@ -414,61 +418,78 @@ export default class ImportDataset {
             folderId
         };
 
-        this.processRequest = Hoot.api.uploadDataset( data )
-            .then( resp => {
-                this.loadingState();
-
-                this.jobId = resp.data[ 0 ].jobid;
-
-                return Hoot.api.statusInterval( this.jobId );
-            } )
-            .then( async resp => {
-                let message;
-                if (resp.data && resp.data.status === 'cancelled') {
-                    message = 'Import job cancelled';
-
-                    // Delete the newly created folder
-                    if ( newFolderName ) {
-                        await Hoot.api.deleteFolder( folderId );
-                        await Hoot.folders.removeFolder( folderId );
-                    }
-
-                    this.submitButton
-                        .select( 'span' )
-                        .text( 'Import' );
-                } else {
-                    message = 'Import job complete';
+        this.loadingState();
+
+        this.processRequest =  Hoot.api.uploadDataset( data, this.source )
+        .then( resp => {
+            this.jobId = resp.data[ 0 ].jobid;
+            return Hoot.api.statusInterval( this.jobId );
+        } )
+        .then( async resp => {
+            let message;
+
+            if ( resp.message === 'Request canceled'  ) {
+                Hoot.message.alert( {
+                    message: resp.message,
+                    type: 'warn'
+                } );
+            }
+
```

Comment:
I'm not sure we need this both here and below.  Seems like we call `source.cancel()` with "Request cancelled" with two Ls so not sure this ever gets hit.